[DOC] Create a suite of pages for Explore Traces (#235)
grafana/explore-traces: docs/sources/investigate/index.md

canonical: https://grafana.com/docs/grafana/latest/explore/simplified-exploration/traces/investigate/
keywords:
  - Explore Traces
  - Investigate
title: Investigate trends and spikes
menuTitle: Investigate trends and spikes
weight: 600
---

# Investigate trends and spikes

{{< docs/public-preview product="Explore Traces" >}}

Explore Traces provides powerful tools that help you identify and analyze problems in your applications and services.

Using these steps, you can use the tracing data to investigate issues.

1. Select the primary signal.
1. Choose the metric you want to use: rates, errors, or duration.
1. Define filters to refine the view of your data.
1. Use the structural or trace list to drill down into the issue.

{{< docs/play title="the Grafana Play site" url="https://play.grafana.org/a/grafana-exploretraces-app/explore" >}}

## Change the primary signal type to observe

Tracing data is highly structured and annotated and reflects events that happen in your services.
You can choose the type of services you want to observe and think about.

You can use the full trace (trace roots), or you can select a more specific type, such as service or database calls.

> By default, Explore Traces displays information about full traces. You can change this by using the selector in the Filter bar.

Reviewer: Is "Filter bar" part of the UI? Do we need bold?

Author: It probably should be bolded. 
```suggestion
By default, Explore Traces displays information about full traces. You can change this by using the selector in the **Filter** bar.
```

![Selecting a signal type](../images/explore-traces-select-signal.png)

You can use any one of these primary signal types.

Full traces
: Inspect full journeys of requests across services

Server spans
: Explore service-specific segments of traces

Consumer spans
: Analyze how queues are consumed. A consumer span indicates an asynchronous handoff between services. These handoffs are almost always a queue.

Database calls
: Evaluate performance issues in database interactions

All spans
: View and analyze raw span data

> 

Author: This area is a UI explanation. The order and list match the UI menu. I'm not sure about changing it around here since we're explaining the UI specifically.

## Choose a RED metric

Explore Traces uses RED metrics generated from your tracing data to guide your investigation.
In this context, RED metrics mean:

* **Rates** show the rate of incoming spans per second.
* **Errors** show spans that are failing.
* **Duration** displays the amount of time those spans take; represented as a heat map that shows response time and latency.

When you select a RED metric, the tabs underneath the metrics selection changes match the context.
For example, selecting **Duration** displays **Root cause latency** and **Slow traces tabs**.
Choosing **Errors** changes the tabs to **Root cause errors** and **Errored traces**. Rate provides **Service structure**, and **Traces** tabs.
These tabs are used when you [analyze tracing data](#analyze-tracing-data).

To choose a RED metric:

> 1. Select a metric type a graph to select a **Rate**, **Errors**, or **Duration** metric type. Notice that your selection changes the first drop-down list on the filter bar.

Reviewer 2: ```suggestion
1. Select a graph to select a **Rate**, **Errors**, or **Duration** metric type. Notice that your selection changes the first drop-down list on the filter bar.
```

This seems like it might be a typo.

When you're using **Duration** metrics, **Breakdown** orders the sequence of attributes by their average duration.
When you select **Rate**, **Breakdown** orders the sequence of attributes by their rate of requests per second, with errors colored red.

You can change the **Scope** to show **Resource** or **Span**.

Using the **Group by** selector, you can group the selected metric by different attributes.
For example, if you have selected **Errors** as a metric type and then choose the `service.name` attribute, the displayed results show the number of errors sorted by the `service.name` with the most matches.

![Errors metric showing the **Breakdown** tab without filters](/media/docs/explore-traces/explore-traces-breakdown-errors-v0.9.png)

The app defaults to `service.name` and displays other commonly used resource level attributes such as `cluster`, `environment`, and `namespace`.
In the drop-down list, you can choose any resource level attribute to group by.

### Modify a filter

Selecting an option for a filter automatically updates the displayed data.
If there are no matches, the app displays a “No data for selected query” message.

To modify an applied filter:

> 1. Select the filter to modify in the Filter bar.

Reviewer: Is Filter bar part of the UI? Do we need bold?

Author: ```suggestion
1. Select the filter to modify in the filter bar.
```

1. Select an option from the drop-down list.

You can also click in the **Filter** bar to add filters using drop-down lists.

> ### Remove one or more filters

Reviewer 2: ```suggestion
### Remove filters
```

Maybe a small nit, I think this can be more concise, and you have instructions for both in the section.

You can remove all or individual filters.

To remove a filter, select **Remove filter** (**X**) at the end of the filter you want to remove.

> To remove all filters, select **Clear filters** (**X**) from the right side of the Filter bar.

Author: ```suggestion
To remove all filters, select **Clear filters** (**X**) from the right side of the filter bar.
```

* Rate provides **Service structure**
* Errors provides **Root cause errors**
* Duration metrics provides **Root cause latency**

For **Rate**, the **Service structure** tab shows you how your applications "talk" to each other to fulfill requests.
Use this tab to analyze the service structure of the traces that match the current filters.

![Service structure tab](/media/docs/explore-traces/explore-traces-rate-service-structure-v0.9.png)

For **Errors**, the **Root cause errors** tab shows structure of errors beneath your selected filters. Use this tab to immediately see the chain of errors that are causing issues higher up in traces.

![Link to span data from Root cause errors](/media/docs/explore-traces/explore-traces-errors-root-cause-v0.9.png)

When you select **Duration** metrics, the **Root cause latency** tab shows the structure of the longest running spans so you can analyze the structure of slow spans.

The pictured spans are an aggregated view compiled using spans from multiple traces.

![Duration metric showing root cause latency](/media/docs/explore-traces/explore-traces-duration-root-cause-latency.png)

### Trace list

Each RED metric has a trace list:

> * **Rate** provides a tab that lists **Traces**
> * **Errors** provides a list of **Errored traces**
> * **Duration** lists **Slow traces**

Author: ```suggestion
* **Rate** provides a tab that lists **Traces**.
* **Errors** provides a list of **Errored traces**.
* **Duration** lists **Slow traces**.
```

![Example trace list for Duration showing slow traces](/media/docs/explore-traces/explore-traces-duration-slow-traces-v0.9.png)

## Change selected time range

Use the time picker at the top right to modify the data shown in Explore Traces.

> You can select a time range of up to 24h hours in duration.
> This time range can be any 24h period in your configured trace data retention period.

Reviewer 2: ```suggestion
You can select a time range of up to 24 hours in duration.
This time range can be any 24-hour period in your configured trace data retention period.
```
